### PR TITLE
[Chat] Identity package

### PR DESCRIPTION
**Pull request by flypaper0**
# Description
- WalletConnectIdentity package
- JWTFactory restructured
- Web3Inbox debug session

Resolves # (issue)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Chat/ProtocolServices/Common/MessagingService.swift [hunk 1]
```diff
@@ -4,9 +4,8 @@ import Combine
 class MessagingService {
 
     private let keyserverURL: URL
-    private let networkingInteractor: NetworkInteracting
-    private let identityStorage: IdentityStorage
-    private let identityService: IdentityService
+    private let networkClient: NetworkInteracting
```
Comment (llbartekll) [inline Sources/Chat/ProtocolServices/Common/MessagingService.swift:7]: should we rename networkingInteractor to networkClient? we already got use to `networkingInteractor` and it's consistent in all the SDKs

I personally wouldn't do it

File: Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift [hunk 2]
```diff
@@ -104,46 +108,33 @@ private extension InvitationHandlingService {
...
-                        message: decoded.message,
+                        message: invite.message,
                         inviterAccount: inviterAccount,
-                        inviteeAccount: inviteeAccount,
-                        inviterPublicKey: decoded.publicKey,
+                        inviteeAccount: invite.inviteeAccount,
+                        inviterPublicKey: inviterPublicKey,
                         inviteePublicKey: inviteePublicKey,
-                        timestamp: decoded.iat // TODO: Replace with relay message receivedAt
+                        timestamp: claims.iat // TODO: Replace with relay message receivedAt
```
Comment (llbartekll) [inline Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift:134]: >TODO: Replace with relay message receivedAt
do we plan to have `receivedAt` does it even make sense?
we know know when we receive message and we know when it was sent with publishedAt

Comment (llbartekll) [inline Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift:134]: >TODO: Replace with relay message receivedAt

do we plan to have `receivedAt` does it even make sense?
we know know when we receive message and we know when it was sent with publishedAt

File: Sources/WalletConnectIdentity/IdentityClient.swift [hunk 3]
```diff
@@ -1,62 +1,88 @@
...
+        try await networkClient.subscribe(topic: topic)
     }
 
     func unsubscribeFromInvites(inviteKey: AgreementPublicKey) {
         let topic = inviteKey.rawRepresentation.sha256().toHexString()
         kms.deletePublicKey(for: topic)
-        networkingInteractor.unsubscribe(topic: topic)
+        networkClient.unsubscribe(topic: topic)
     }
 }
```
Comment (llbartekll) [inline Sources/WalletConnectIdentity/IdentityClient.swift:88]: could this be a part of the Chat sdk?
Identity Package for client->server interaction

but this part is just chat peer to peer

File: Sources/Chat/ProtocolServices/Common/MessagingService.swift (hunk 1 above)
Comment (flypaper0) [reply Sources/Chat/ProtocolServices/Common/MessagingService.swift:7]: reverted, planning to discuss it later

File: Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift (hunk 2 above)
Comment (flypaper0) [reply Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift:134]: I meant publishedAt

**Comment by flypaper0** [comment]
@llbartekll suggestions fixed in https://github.com/WalletConnect/WalletConnectSwiftV2/pull/735

**Comment by flypaper0** [comment]
Merged with https://github.com/WalletConnect/WalletConnectSwiftV2/pull/735

**Pull request closed without merging**